File: app/src/main/java/com/lifesolutions/bd/Adapters/GroupChatListAdapter.java
//package com.stardigiinternational.starnotee.Adapters;
//
//import android.content.Context;
//import android.content.Intent;
//import android.view.LayoutInflater;
//import android.view.View;
//import android.view.ViewGroup;
//import android.widget.RelativeLayout;
//import android.widget.TextView;
//import androidx.annotation.NonNull;
//import androidx.recyclerview.widget.RecyclerView;
//
//import com.google.firebase.database.DataSnapshot;
//import com.google.firebase.database.DatabaseError;
//import com.google.firebase.database.DatabaseReference;
//import com.google.firebase.database.FirebaseDatabase;
//import com.google.firebase.database.ValueEventListener;
//import com.squareup.picasso.Picasso;
//import com.stardigiinternational.starnotee.Activities.GroupChatActivity;
//import com.stardigiinternational.starnotee.Models.GroupChat;
//import com.stardigiinternational.starnotee.Models.Message;
//import com.stardigiinternational.starnotee.Models.User;
//import com.stardigiinternational.starnotee.R;
//import java.util.ArrayList;
//import de.hdodenhof.circleimageview.CircleImageView;
//
//public class GroupChatListAdapter extends RecyclerView.Adapter<GroupChatListAdapter.GChatListViewHolder>{
//
//    private Context mContext;
//    private ArrayList<GroupChat> groupChats;
//    private String lastMessage,messageType;
//    private Boolean isSeen = false;
//
//    public GroupChatListAdapter(Context mContext, ArrayList<GroupChat> groupChats) {
//        this.mContext = mContext;
//        this.groupChats = groupChats;
//    }
//
//
//    @NonNull
//    @Override
//    public GroupChatListAdapter.GChatListViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
//        return new GroupChatListAdapter.GChatListViewHolder(LayoutInflater.from(mContext).inflate(R.layout.user_layout,parent,false));
//    }
//
//    @Override
//    public void onBindViewHolder(@NonNull final GroupChatListAdapter.GChatListViewHolder holder, final int position) {
//
//        holder.userName.setText(groupChats.get(position).getGroupTitle());
//        holder.onlineStatus.setVisibility(View.GONE);
//        if (groupChats.get(position).getGroupIcon().equals("none")){
//            holder.user_image.setImageResource(R.drawable.user_low);
//        } else {
//            Picasso.get().load(groupChats.get(position).getGroupIcon()).into(holder.user_image);
//        }
//
//        holder.address.setText(groupChats.get(position).getGroupDescription());
//
//        holder.item.setOnClickListener(view -> mContext.startActivity(new Intent(mContext, GroupChatActivity.class).putExtra("groupId",groupChats.get(position).getGroupId())));
//        lastMessage(groupChats.get(position).getGroupId(),holder.address);
//    }
//
//    @Override
//    public int getItemCount() {
//        return groupChats.size();
//    }
//
//    public class GChatListViewHolder extends RecyclerView.ViewHolder{
//
//        TextView userName,address;
//        CircleImageView user_image,onlineStatus;
//        RelativeLayout item;
//
//        public GChatListViewHolder(@NonNull View itemView) {
//            super(itemView);
//
//            user_image = itemView.findViewById(R.id.profile_image_user);
//            userName = itemView.findViewById(R.id.name_users_item);
//            address = itemView.findViewById(R.id.address_users_item);
//            item = itemView.findViewById(R.id.users_item);
//            onlineStatus = itemView.findViewById(R.id.online_status_user_layout);
//        }
//    }
//
//    private void lastMessage(String groupId,TextView last_msg){
//        lastMessage = "default";
//        DatabaseReference reference = FirebaseDatabase.getInstance().getReference();
//
//
//        reference.child("GroupMessages").child(groupId).addValueEventListener(new ValueEventListener() {
//            @Override
//            public void onDataChange(@NonNull DataSnapshot dataSnapshot) {
//                for (DataSnapshot snapshot : dataSnapshot.getChildren()){
//                    Message message = snapshot.getValue(Message.class);
//                    lastMessage = message.getMessage();
//                    messageType = message.getType();
//                    isSeen = message.isSeen();
//                }
//
//                if ("default".equals(lastMessage)) {
//
//                    last_msg.setText("No messages");
//
//                } else {
//                    if (messageType.equals("image")) {
//
//                        last_msg.setText("Image");
//                        if (isSeen){
//                            last_msg.setTextColor(mContext.getResources().getColor(R.color.text_color_light));
//                        }else {
//                            last_msg.setTextColor(mContext.getResources().getColor(R.color.text_color_light));
//                        }
//
//                    } else if (messageType. equals("pdf")){
//
//                        last_msg.setText("PDF File");
//                        if (isSeen){
//                            last_msg.setTextColor(mContext.getResources().getColor(R.color.text_color_light));
//                        }else {
//                            last_msg.setTextColor(mContext.getResources().getColor(R.color.text_color_light));
//                        }
//
//                    }else if (messageType.equals("docx")){
//
//                        last_msg.setText("DOCX File");
//                        if (isSeen){
//                            last_msg.setTextColor(mContext.getResources().getColor(R.color.text_color_light));
//                        }else {
//                            last_msg.setTextColor(mContext.getResources().getColor(R.color.text_color_light));
//                        }
//
//                    }else if (messageType.equals("voice")){
//
//                        last_msg.setText("Voice Message");
//                        if (isSeen){
//                            last_msg.setTextColor(mContext.getResources().getColor(R.color.text_color_light));
//                        }else {
//                            last_msg.setTextColor(mContext.getResources().getColor(R.color.text_color_light));
//                        }
//
//                    }else {
//
//                        last_msg.setText(lastMessage);
//                        if (isSeen){
//                            last_msg.setTextColor(mContext.getResources().getColor(R.color.text_color_light));
//                        }else {
//                            last_msg.setTextColor(mContext.getResources().getColor(R.color.text_color_light));
//                        }
//                    }
//                }
//
//                lastMessage = "default";
//            }
//
//            @Override
//            public void onCancelled(@NonNull DatabaseError databaseError) {
//
//            }
//        });
//    }
//}

File: app/src/main/java/com/lifesolutions/bd/kotlinCode/ui/adapter/GroupChatListAdapter.kt
package com.lifesolutions.bd.kotlinCode.ui.adapter

import android.annotation.SuppressLint
import android.content.Context
import android.content.Intent
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.ImageView
import android.widget.TextView
import androidx.constraintlayout.widget.ConstraintLayout
import androidx.recyclerview.widget.RecyclerView
import com.squareup.picasso.Picasso
import com.lifesolutions.bd.R
import com.lifesolutions.bd.kotlinCode.data.model.GroupConversation
import com.lifesolutions.bd.kotlinCode.ui.activities.GroupMessageActivity
import kotlinx.android.synthetic.main.activity_join_group.*
import kotlinx.android.synthetic.main.group_chat_list_item.view.*
import java.text.SimpleDateFormat
import java.util.*

class GroupConversationListAdapter(
    private val context: Context
) : RecyclerView.Adapter<GroupConversationListAdapter.ListViewAdapter>() {

    private var groupList = mutableListOf<GroupConversation>()
    val  userPreferences = context.getSharedPreferences("CurrentUser", Context.MODE_PRIVATE)!!

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ListViewAdapter {
        return ListViewAdapter(
            LayoutInflater.from(parent.context)
                .inflate(R.layout.group_chat_list_item, parent, false)
        )
    }

    override fun getItemCount() = groupList.size

    @SuppressLint("SimpleDateFormat")
    override fun onBindViewHolder(holder: ListViewAdapter, position: Int) {
        val group = groupList[position]

        holder.groupName.text = group.groupName

        // val sdf = SimpleDateFormat("dd/MM/yy hh:mm a")
        val sdf = SimpleDateFormat("dd/MM/yy")
        val netDate = Date(group.createdTime!!)
        val date = sdf.format(netDate)
        holder.lastConversation.text = group.lastConversation.take(35)

        // Set Profile Image
        if (group.groupImage != null && group.groupImage != "none") {
            Picasso.get().load(group.groupImage).into(holder.groupImage)
        } else {
            Picasso.get().load(R.drawable.ic_group_placeholder).into(holder.groupImage)
        }



        // Click..
        holder.conversationLayout.setOnClickListener {
            Intent(context, GroupMessageActivity::class.java).apply {
                putExtra("groupId", group.groupId)
                context.startActivity(this)
            }
        }

        // checkCurrentUserRole(group.groupId!!, holder)

    }

    /**
     * Extra Functions
     */
    fun addAllConversation(newConversation: ArrayList<GroupConversation>) {
        groupList = newConversation
        notifyDataSetChanged()
    }

    fun removeAllItem() {
        groupList.clear()
    }


//    private fun checkRequestList(groupId: String, holder: ListViewAdapter) {
//        val ref = FirebaseDatabase.getInstance().getReference("GroupConversation").child(groupId)
//
//            ref.child("Requests").addValueEventListener(object: ValueEventListener{
//                @SuppressLint("SetTextI18n")
//                override fun onDataChange(dataSnapshot: DataSnapshot) {
//                    if (dataSnapshot.exists()) {
//                       val reqCount = dataSnapshot.childrenCount
//                        holder.requestCount.text = reqCount.toString()
//                    } else {
//                        holder.requestCount.visibility = View.GONE
//                    }
//                }
//
//                override fun onCancelled(p0: DatabaseError) {
//                    TODO("Not yet implemented")
//                }
//
//
//            })
//    }

//    private fun checkCurrentUserRole(groupId: String, holder: ListViewAdapter) {
//        val authId = userPreferences.getString("uID", null)
//        val ref = FirebaseDatabase.getInstance().getReference("GroupConversation").child(groupId)
//
//        ref.child("Members").child(authId!!).addListenerForSingleValueEvent(object: ValueEventListener{
//            @SuppressLint("SetTextI18n")
//            override fun onDataChange(dataSnapshot: DataSnapshot) {
//                if (dataSnapshot.exists()) {
//                    val member = dataSnapshot.getValue(GroupMember::class.java)
//                    if (member?.role == "admin") {
//                        checkRequestList(groupId, holder)
//                    }
//                } else {
//                    holder.requestCount.visibility = View.GONE
//                }
//            }
//
//            override fun onCancelled(p0: DatabaseError) {
//                TODO("Not yet implemented")
//            }
//
//
//        })
//    }


    inner class ListViewAdapter(itemView: View) : RecyclerView.ViewHolder(itemView) {
        val conversationLayout: ConstraintLayout = itemView.constraint_layout_group_list
        val groupImage: ImageView = itemView.iv_profile_user_group_list
        val groupName: TextView = itemView.tv_group_name_group_list
        val requestCount: TextView = itemView.tv_request_count
        val lastConversation: TextView = itemView.tv_last_msg_group_list
    }

}